### PR TITLE
Link App notifications settings + screenshots (EXPOSUREAPP-9618, 9617)

### PR DESCRIPTION
**Pull request by jurajkusnier**
## New notifications + screenshots

device-2021-10-04-095923

## Testing
- check if notifications work as expected

## Jira Story
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9497

### CODE REVIEW

**Review by AlexanderAlferov — DISMISSED** [review]
Tested with Samsung S8: worked 🚀

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingDeltaNotificationsFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            object : OnboardingDeltaNotificationsViewModel.Factory {
+                override fun create(): OnboardingDeltaNotificationsViewModel =
+                    OnboardingDeltaNotificationsViewModel(
+                        settings = settings,
+                        dispatcherProvider = TestDispatcherProvider(),
+                        analyticsSettings = analytics
+                    )
+            }
+        )
+    }
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingDeltaNotificationsFragmentTest.kt:40]: you can just mock the viewModel 
```kotlin
    @MockK lateinit var viewModel: OnboardingDeltaNotificationsViewModel

    @Before
    fun setup() {
        MockKAnnotations.init(this, relaxed = true)

        setupMockViewModel(
            object : OnboardingDeltaNotificationsViewModel.Factory {
                override fun create(): OnboardingDeltaNotificationsViewModel = viewModel
            }
        )
```

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/settings/notifications/NotificationSettingsFragmentTest.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        setupMockViewModel(
+            object : NotificationSettingsFragmentViewModel.Factory {
+                override fun create(): NotificationSettingsFragmentViewModel =
+                    NotificationSettingsFragmentViewModel(
+                        notificationSettings = settings,
+                        dispatcherProvider = TestDispatcherProvider(),
+                        generalNotifications = generalNotifications,
+                        presenceTracingNotifications = presenceTracingNotifications,
+                        digitalCovidCertificateNotifications = digitalCovidCertificateNotifications,
+                    )
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/settings/notifications/NotificationSettingsFragmentTest.kt:43]: Same here , you can just mock the viewModel

File: Corona-Warn-App/src/androidTest/java/testhelpers/FragmentTestModuleRegistrar.kt [hunk 3]
```diff
@@ -119,6 +122,9 @@ import de.rki.coronawarnapp.ui.vaccination.CovidCertificateInfoFragmentTestModul
         RATProfileQrCodeFragmentTestModule::class,
 
         QrCodeScannerFragmentTestModule::class,
+
+        // --------- Settings ---------
+        NotificationSettingsFragmentModule::class,
```
Comment (mtwalli) [inline Corona-Warn-App/src/androidTest/java/testhelpers/FragmentTestModuleRegistrar.kt:127]: Please call it `NotificationSettingsFragmentTestModule` ? to distinguish it from actual module

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/notifications/NotificationSettingsFragmentViewModel.kt [hunk 4]
```diff
@@ -21,6 +30,14 @@ class NotificationSettingsFragmentViewModel @AssistedInject constructor(
         .map { NotificationSettingsState(it) }
         .asLiveData(dispatcherProvider.Default)
 
+    fun createNotificationChannels() {
+        if (BuildVersionWrap.hasAPILevel(Build.VERSION_CODES.O)) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/notifications/NotificationSettingsFragmentViewModel.kt:34]: We can get rid of these checks when we rely on `NotificationManagerCompat`, `presenceTracingNotifications, digitalCovidCertificateNotifications` are already using it ,but `generalNotifications`
Look at https://github.com/corona-warn-app/cwa-app-android/pull/4169

**Comment by mtwalli** [comment]
Screenshot 2021-10-04 at 18 57 57
- back arrow 🔙  is NOT working

**Comment by jurajkusnier** [comment]
> - back arrow 🔙 is NOT working

FIXED 🛠️

**Review by mtwalli — APPROVED** [review]
LGTM

**Review by AlexanderAlferov — APPROVED** [review]

**Pull request merged**